init.sh: move ipip device creation to Go

rgo3 opened the pull request:
vendor: update vishvananda/netlink:
```
This pulls in the following recent changes:

- vishvananda/netlink@88c0de
- vishvananda/netlink@16d31d

These are needed to move ipip device creation from init.sh to Go.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>
```
init.sh: move ipip device creation to Go:
```
This refactors the creation of the cilium_ipip{4,6} devices to a
Go based implementation and adds a simple device creation test.

One behavioural change was made: no longer try to repurpose
tunl0 by renaming it and enabling collect_md, as that dance is
rather tricky and fragile due to the kernel module needing to be
loaded first.

With this change, we still prefix the fallback devices with cilium_,
meaning cilium_tunl0 for v4 and cilium_ip6tnl0 for v6, but we no
longer try to use or reconfigure these devices. cilium_ipip4/6 are
always created with the correct parameters from the start, reducing
the amount of moving parts.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>
```
Fixes: #20740

rgo3 committed:
init.sh: move ipip device creation to Go

This refactors the creation of the cilium_ipip{4,6} devices to a
Go-based implementation and adds a simple device creation test.

One behavioural change was made: no longer try to repurpose
tunl0 by renaming it and enabling collect_md, as that dance is
rather tricky and fragile due to the kernel module needing to be
loaded first.

With this change, we still prefix the fallback devices with cilium_,
meaning cilium_tunl0 for v4 and cilium_ip6tnl0 for v6, but we no
longer try to use or reconfigure these devices. cilium_ipip4/6 are
always created with the correct parameters from the start, reducing
the amount of moving parts.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>
Co-authored-by: Timo Beckers <timo@isovalent.com>

rgo3 committed:
vendor: update vishvananda/netlink

This pulls in the following recent changes:

- vishvananda/netlink@88c0defd466a5d755d6c3956617ae27217bdda87
- vishvananda/netlink@16d31db2358884ece09bbbaa2b754c225c788032

These are needed to move ipip device creation from init.sh to Go.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>

ti-mo commented on pkg/datapath/loader/base.go:
```diff
@@ -317,11 +317,6 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		mode = tunnelMode
 		encapProto = option.Config.TunnelProtocol
-	case option.Config.EnableHealthDatapath:
-		mode = option.DSRDispatchIPIP
```
This statement has gone missing in this PR. I'll push my local branch with this readded in a bit.

rgo3 replied:
I figured it's no longer needed given it sets a variable that's used as the input arg for init.sh with a value that's no longer used in init.sh after this PR.

ti-mo replied:
Ah! I figured as much, thanks for the context. I'll leave it in, and we'll remove all of them when init.sh is dead. :+1:

lmb commented on pkg/datapath/loader/netlink.go:
```diff
@@ -444,3 +446,127 @@ func addHostDeviceAddr(hostDev netlink.Link, ipv4, ipv6 net.IP) error {
+//   - Rename sit0 to cilium_sit, if present. This was potentially a mistake,
+//     as the sit module is not involved with ip6tnl interfaces.
+//
+// As of Cilium 1.15, if present, tunl0 is renamed to cilium_tunl and ip6tnl0 is
```
This could use a rationale why we're doing this, and at which point we can stop doing this (if ever).

ti-mo replied:
Good point, added.

lmb commented on pkg/datapath/loader/netlink.go:
```diff
@@ -444,3 +446,127 @@ func addHostDeviceAddr(hostDev netlink.Link, ipv4, ipv6 net.IP) error {
+
+		// Rename fallback device created by potential kernel module load after
+		// creating tunnel interface.
+		if err := renameDevice("tunl0", "cilium_tunl"); err != nil {
```
Meta: we should stop relying on device names so much and instead track indices. That saves us from renames breaking stuff.

ti-mo committed:
.github: roll back bpf-next lvh image

Due to the way we're currently tagging lvh images and how that interacts
with renovate, the current image doesn't have the latest changes made to
the build config. Manually drop the pin until a new round of images is built.

Signed-off-by: Timo Beckers <timo@isovalent.com>
